net-mgmt/pfSense-pkg-lldpd/files/usr/local/www/lldpd_settings.php

```diff
@@ -0,0 +1,347 @@
+$section->add($group);
+
+$group = new Form_Group('Extreme Discovery Protocol (EDP)');
+$group->add(new Form_Select(
+	'edp_proto',
+	null,
+	$pconfig['edp_proto'],
+	$protocol_options
+));
+$group->addClass('proto');
+$section->add($group);
+
+$group = new Form_Group('Foundry Discovery Protocol (FDP)');
+$group->add(new Form_Select(
+	'fdp_proto',
+	null,
+	$pconfig['fdp_proto'],
+	$protocol_options
+));
+$group->addClass('proto');
+$section->add($group);
+
+$group = new Form_Group('Nortel Discovery Protocol (NDP)');
+$group->add(new Form_Select(
+	'ndp_proto',
+	null,
+	$pconfig['ndp_proto'],
+	$protocol_options
+));
+$group->addClass('proto');
+$section->add($group);
+
+/* Receive only variants of protocols */
+
+$group = new Form_Group('Link Layer Discover Protocol (LLDP)');
+$group->add(new Form_Select(
+	'lldp_proto_ro',
+	null,
+	$pconfig['lldp_proto'],
+	$protocol_options_ro
+));
+$group->addClass('proto_ro');
+$section->add($group);
+
+$group = new Form_Group('Cisco Discovery Protocol (CDP)');
+$group->add(new Form_Select(
+	'cdp_proto_ro',
+	null,
+	$pconfig['cdp_proto'],
+	$protocol_options_cdp_ro
+));
+$group->addClass('proto_ro');
+$section->add($group);
+
+$group = new Form_Group('Extreme Discovery Protocol (EDP)');
+$group->add(new Form_Select(
+	'edp_proto_ro',
+	null,
+	$pconfig['edp_proto'],
+	$protocol_options_ro
+));
+$group->addClass('proto_ro');
+$section->add($group);
+
+$group = new Form_Group('Foundry Discovery Protocol (FDP)');
+$group->add(new Form_Select(
+	'fdp_proto_ro',
+	null,
+	$pconfig['fdp_proto'],
+	$protocol_options_ro
+));
+$group->addClass('proto_ro');
+$section->add($group);
+
+$group = new Form_Group('Nortel Discovery Protocol (NDP)');
+$group->add(new Form_Select(
+	'ndp_proto_ro',
+	null,
+	$pconfig['ndp_proto'],
+	$protocol_options_ro
+));
+$group->addClass('proto_ro');
+$section->add($group);
+
+
+$form->add($section);
+print($form);
+
+?>
+
+<script type="text/javascript">
+//<![CDATA[
+events.push(function() {
+	function map_proto_ro(value) {
+		if (value == 'disabled') {
+			return value;
+		} else {
+			return 'passive';
+		}
+	}
+
+	function map_proto_cdp_ro(value) {
+		if (value == 'disabled' || value == 'passive' || value == 'passive_v2_only') {
+			return value;
+		} else if (value == 'active_v2_only') {
+			return 'passive_v2_only';
+		} else {
+			return 'passive';
+		}
+	}
+
+	function receiveonlyChange(enabled) {
+		if (enabled) {
+			document.getElementById('lldp_proto_ro').value = map_proto_ro(document.getElementById('lldp_proto').value);
+			document.getElementById('cdp_proto_ro').value = map_proto_cdp_ro(document.getElementById('cdp_proto').value);
+			document.getElementById('edp_proto_ro').value = map_proto_ro(document.getElementById('edp_proto').value);
+			document.getElementById('fdp_proto_ro').value = map_proto_ro(document.getElementById('fdp_proto').value);
+			document.getElementById('ndp_proto_ro').value = map_proto_ro(document.getElementById('ndp_proto').value);
+
```

Comment:
This type of construct has mostly been eliminated from pfSense in favor of jQuery. E.g.:

`$('#lldp_proto_ro').val($('#lldp_proto').val());`

Reply:
Okay, thanks.